test/FxERC20.js

```diff
@@ -0,0 +1,209 @@
+
+            // Tokens must be locked on the FxERC20ChildTunnel contract address
+            const balance = await childToken.balanceOf(fxERC20ChildTunnel.address);
+            expect(balance).to.equal(amount);
+
+            // On L1 fxERC20RootTunnel will be passed a proof validation data that the tx has happened on L2
+            await rootToken.mint(deployer.address, amount);
+        });
+
+        it("Deposit tokens to a different address", async function () {
+            const account = signers[1].address;
+
+            // Approve tokens
+            await childToken.approve(fxERC20ChildTunnel.address, amount);
+
+            // Send tokens to L1
+            await fxERC20ChildTunnel.connect(deployer).depositTo(account, amount);
+
+            // Tokens must be locked on the FxERC20ChildTunnel contract address
+            const balance = await childToken.balanceOf(fxERC20ChildTunnel.address);
+            expect(balance).to.equal(amount);
+
+            // On L1 fxERC20RootTunnel will be passed a proof validation data that the tx has happened on L2
+            await rootToken.mint(account, amount);
+        });
+
+        it("Withdraw tokens", async function () {
+            // Approve tokens
+            await childToken.connect(deployer).approve(fxERC20ChildTunnel.address, amount);
+
+            // Send tokens to L1
+            await fxERC20ChildTunnel.connect(deployer).deposit(amount);
+
+            // Tokens must be locked on the FxERC20ChildTunnel contract address
+            let balance = await childToken.balanceOf(fxERC20ChildTunnel.address);
+            expect(balance).to.equal(amount);
+
+            // On L1 fxERC20RootTunnel will be passed a proof validation data that the tx has happened on L2
+            await rootToken.mint(deployer.address, amount);
+
+            // Withdraw tokens
+            await rootToken.connect(deployer).approve(fxERC20RootTunnel.address, amount);
+
+            // Root token must be owned by the FxERC20RootTunnel contract
+            await rootToken.changeOwner(fxERC20RootTunnel.address);
+
+            // Burn tokens on L1 and send message to L2 to retrieve them there
+            await expect(
+                fxERC20RootTunnel.withdraw(amount)
+            ).to.be.reverted;
+
+            // Get the message on L2
+            const data = ethers.utils.solidityPack(
+                ["address", "address", "uint256"],
+                [deployer.address, deployer.address, amount]
+            );
+
+            // Upon message receive, tokens on L2 are transferred to the destination account (deployer)
+            await fxERC20ChildTunnel.connect(deployer).processMessageFromRoot(stateId, fxERC20RootTunnel.address, data);
+
+            // There must be no balance left locked on the FxERC20ChildTunnel contract
+            balance = await childToken.balanceOf(fxERC20ChildTunnel.address);
+            expect(balance).to.equal(0);
```

Comment:
where we check that the receiver has non-zero balance?

Reply:
was just lazy to check in one place :P